s3box_review/README.md

```diff
@@ -0,0 +1,126 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [S3Box](#s3box)
+  - [S3Box - The Configuration](#s3box---the-configuration)
+    - [NewS3Box](#news3box)
+    - [NewS3BoxOptions](#news3boxoptions)
+  - [S3Box - The Methods](#s3box---the-methods)
+    - [Pack](#pack)
+    - [Seal](#seal)
+    - [CreateManifests](#createmanifests)
+    - [NextBox](#nextbox)
+- [Example](#example)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# S3Box
+
+Library aiding data transport to s3 through straighforward configuration and intuitive methods (Pack, Seal and CreateManifest).
+
+The core under-the-hood functionality is the streaming of (JSON formatted) data into s3 while managing consistent file sizes and easy creation of manifests.
+
+See both below and the [the godocs](https://godoc.org/github.com/cgclever/redbox/s3box) for API documentation.
+
+## S3Box - The Configuration
+
+### NewS3Box
+
+`func NewS3Box(options NewS3BoxOptions) (*S3Box, error)`
+
+### NewS3BoxOptions
+
+```
+type NewS3oxOptions struct {
+	// Required inputs
+	S3Bucket          string
+
+  // Optional AWS creds. If not provided they'll be grabbed from the environment.
+	AWSKey            string
+	AWSPassword       string
+	AWSToken          string
+	
+	// Optional
+	BufferSize  int    // Default 10MB
+}
+```
+
+The expected usecase around controlling BufferSize is for worker memory management. To comfortably use this package at least `2*BufferSize` of memory should be available at any time.
+
+## S3Box - The Methods
+
+### Pack
+
+`func Pack(data []byte) error`
+
+Pack buffers the data. Once the buffer grows larger than it's maximum size (10MB by default)
+the data is gziped and streamed to s3.
+
+Currently Pack is a single row operation which *only* accepts JSONifiable inputs, i.e. those marshalable into a map.
+
+Pack is concurrency safe.
+
+### Seal
+
+`func Seal() error`
+
+Seal flushes any buffered data to s3 and prevents further Packing.
+
+### CreateManifests
+
+`func CreateManifests(manifestKey string, numManifests int) ([]string, error)`
```

Comment:
I've been wondering the same. The verbiage was intuitive to me that you would "Pack and Seal". The primary value of "Seal" would be to protect yourself to ensure no more writes can happen.